utils/lit/lit/formats/taef.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import absolute_import
+import os
+import sys
+
+import lit.Test
+import lit.TestRunner
+import lit.util
+from .base import TestFormat
+
+kIsWindows = sys.platform in ['win32', 'cygwin']
+
+class TaefTest(TestFormat):
+    def __init__(self, te_path, test_dll_path, hlsl_data_dir, test_path):
+        self.te = te_path
+        self.test_dll = test_dll_path
+        self.hlsl_data_dir = hlsl_data_dir
+        self.test_path = test_path
+
+    def getTaefTests(self, litConfig, localConfig):
+        """getTaefTests()
+
+        Return the tests available in taef test dll.
+
+        Args:
+          litConfig: LitConfig instance
+          localConfig: TestingConfig instance"""
+
+        # TE:F:\repos\DxcGitHub\hlsl.bin\TAEF\x64\te.exe
+        # test dll : F:\repos\DxcGitHub\hlsl.bin\Debug\test\clang-hlsl-tests.dll
+        # /list
+
+        try:
+            lines = lit.util.capture([self.te, self.test_dll, '/list'],
+                                     env=localConfig.environment)
+            if kIsWindows:
+              lines = lines.replace('\r', '')
+            lines = lines.split('\n')
+        except:
+            litConfig.error("unable to discover taef in %r" % path)
+            raise StopIteration
+
+        nested_tests = []
```

Comment:
This is copied from the google test implementation (which I get), but this makes no sense here. Googletest does this because the test listing doesn't print the full test name so it need so combine partial strings. Here, the `nested_tests` variable is always an empty array.